presto-hive/src/test/java/com/facebook/presto/hive/TestHiveMaterializedViewUtils.java

```diff
@@ -233,6 +233,54 @@ public void testDifferenceDataPredicates()
         comparePredicates(diffDataPredicates, keys, partitionResults.build());
     }
 
+    @Test
+    public void testDifferenceDataOuterJoin()
+    {
+        TestingTypeManager typeManager = new TestingTypeManager();
+        TestingSemiTransactionalHiveMetastore testMetastore = TestingSemiTransactionalHiveMetastore.create();
+
+        List<String> keys = ImmutableList.of("ds");
+        Column dsColumn = new Column("ds", HIVE_STRING, Optional.empty());
+        List<Column> partitionColumns = ImmutableList.of(dsColumn);
+        List<String> partitions = ImmutableList.of(
+                "ds=2020-01-01",
+                "ds=2020-01-02",
+                "ds=2020-01-03",
+                "ds=2020-01-04",
+                "ds=2020-01-05",
+                "ds=2020-01-06");
+
+        testMetastore.setPartitionNames(partitions);
+
+        MaterializedDataPredicates baseDataPredicates =
+                getMaterializedDataPredicates(testMetastore, metastoreContext, typeManager, getTable(partitionColumns), DateTimeZone.UTC);
+
+        // CREATE MV AS SELECT ds1 as t1.ds, ds2 as t2.ds FROM t1 LEFT JOIN t2 ON t1.ds = t2.ds
+        List<String> viewPartitions = ImmutableList.of(
+                "ds1=2020-01-02/ds2=2020-01-02",
+                "ds1=2020-01-03/ds2=null",
```

Comment:
For the former case, we should invalidate this partition when new t2 partition comes up. I will implement this logic next, on partition update handling side

Reply:
I see. So this differenceDataPredicates() doesn't consider the version of the partitions when doing the diff. So it can only be used to make decisions which do not affect correctness. Let's add this assumption to the function comments.